src/app/charts/line-graph.component.ts

```diff
@@ -132,6 +131,12 @@ export class LineGraphComponent {
         // Time scales only recognize annual and daily data
         var parseTime = D3.timeParse(this.timeFormat);
         this.extractedData.forEach(d => d.date = parseTime(d.date));
+
+        // Sort data by date ascending
+        this.extractedData.sort(function(a, b) {return a.date - b.date;});
```

Comment:
This creates a type checking error:
```
[default] /home/local/AZVA-INT/khoekema/projects/climate/climate-change-lab/src/app/charts/line-graph.component.ts:136:56 
    The left-hand side of an arithmetic operation must be of type 'any', 'number' or an enum type.
```
To me it seems like the problem goes back a little ways.  I.e.

- The `DataPoint` type defines its `date` attribute as a string [here] (https://github.com/azavea/climate-change-lab/blob/feature/SupportMonthlyIndicators/src/app/models/chart.ts#L40)
- But the `LineGraphComponent` overwrites it with a javascript Date object [here] (https://github.com/azavea/climate-change-lab/blob/feature/SupportMonthlyIndicators/src/app/charts/line-graph.component.ts#L133)

That seems like a bad situation.  If we want dates, we should convert much earlier (e.g. [here] (https://github.com/azavea/climate-change-lab/blob/feature/SupportMonthlyIndicators/src/app/services/chart.service.ts#L41) where the data is initially loaded).  That would require moving the config that's used to parse and interpret the date to somewhere where it can be shared, but that might be a good thing anyway.  Anything that's using `DataPoint` values as defined (i.e. treating the dates as strings) would have to be checked/revised to handle Date objects.

If we mostly want to keep the date keys as returned by the API (i.e. 'YYYY', 'YYYY-MM', or 'YYYY-MM-DD' strings) then we should define a separate data type for use inside `LineGraphComponent` and explicitly convert the data to that format.  (Also, `setLineScales` seems like a weird place to mutate the data in a way that affects all subsequent steps. Doing it in `filterData` where the `extractedData` object is created would make more sense to me.)